cpp/arcticdb/storage/rocksdb/rocksdb_storage.cpp

```diff
@@ -0,0 +1,239 @@
+    auto handle = handles_by_key_type_.at(key_type_name);
+    ::rocksdb::Iterator* it = db_->NewIterator(::rocksdb::ReadOptions(), handle);
+    for (it->SeekToFirst(); it->Valid(); it->Next()) {
+        auto key_slice = it->key();
+        auto k = variant_key_from_bytes(reinterpret_cast<const uint8_t *>(key_slice.data()), key_slice.size(), key_type);
+
+        ARCTICDB_DEBUG(log::storage(), "Iterating key {}: {}", variant_key_type(k), variant_key_view(k));
+        if (prefix_matcher(variant_key_id(k))) {
+            visitor(std::move(k));
+        }
+    }
+    auto s = it->status();
+    util::check(s.ok(), DEFAULT_ROCKSDB_NOT_OK_ERROR + s.ToString());
+    delete it;
+}
+
+std::vector<VariantKey> RocksDBStorage::do_remove_internal(Composite<VariantKey>&& ks, RemoveOpts opts)
+{
+    auto grouper = [](auto &&k) { return variant_key_type(k); };
+    std::vector<VariantKey> failed_deletes;
+
+    (fg::from(ks.as_range()) | fg::move | fg::groupBy(grouper)).foreach([&](auto &&group) {
+        auto key_type_name = fmt::format("{}", group.key());
+        // If no key of this type has been written before, this can fail
+        auto handle = handles_by_key_type_.at(key_type_name);
+        auto options = ::rocksdb::WriteOptions(); // Should this be const class attr? Used in write_internal too.
+        options.sync = true;
+        for (const auto &k : group.values()) {
+            if (do_key_exists(k)) {
+                auto k_str = to_serialized_key(k);
+                auto s = db_->Delete(options, handle, ::rocksdb::Slice(k_str));
+                util::check(s.ok(), DEFAULT_ROCKSDB_NOT_OK_ERROR + s.ToString());
+                ARCTICDB_DEBUG(log::storage(), "Deleted segment for key {}", variant_key_view(k));
+            } else if (!opts.ignores_missing_key_) {
+                log::storage().warn("Failed to delete segment for key {}", variant_key_view(k));
+                failed_deletes.push_back(k);
+            }
+        }
+    });
+    return failed_deletes;
+}
+
+void RocksDBStorage::do_write_internal(Composite<KeySegmentPair>&& kvs)
+{
+    auto grouper = [](auto &&kv) { return kv.key_type(); };
+    (fg::from(kvs.as_range()) | fg::move | fg::groupBy(grouper)).foreach([&](auto &&group) {
+        auto key_type_name = fmt::format("{}", group.key());
+        auto handle = handles_by_key_type_.at(key_type_name);
+        for (auto &kv : group.values()) {
+            auto options = ::rocksdb::WriteOptions();
+            options.sync = true;
+            auto k_str = to_serialized_key(kv.variant_key());
+
+            auto& seg = kv.segment();
+            auto hdr_sz = seg.segment_header_bytes_size();
+            auto total_sz = seg.total_segment_size(hdr_sz);
+            std::string seg_data;
+            seg_data.resize(total_sz);
+            seg.write_to(reinterpret_cast<std::uint8_t *>(seg_data.data()), hdr_sz);
+            auto override = std::holds_alternative<RefKey>(kv.variant_key());
```

Comment:
Different name please, `override` is a language keyword and at the very least breaks the syntax highlighting